src/components/molecules/event/eventRegistrationQR/EventRegistrationQR.tsx

```diff
@@ -0,0 +1,70 @@
+import React from 'react';
+import { Event } from 'models/apiModels';
+import Button from 'components/atoms/button/Button';
+
+import { createQR, getQR } from 'api';
+import { useToast } from 'hooks/useToast';
+
+const EventRegistrationQR: React.FC<{ event: Event }> = ({ event }) => {
+  const { addToast } = useToast();
+
+  const handleDownload = (blob: Blob) => {
+    const filename = event.title + '.pdf';
+    const pdf_url = window.URL.createObjectURL(new Blob([blob]));
+    const link = document.createElement('a');
+    link.href = pdf_url;
+    link.setAttribute('download', filename);
+    link.click();
+  };
+
+  const handleButtonClick = async (url: string) => {
+    /* Try to get QR first */
+    let res;
+    try {
+      res = await getQR(url);
+    } catch (error) {
+      if (error.statusCode !== 400) {
+        addToast({
+          title: 'Feilmelding',
+          status: 'error',
+          description: 'Ukjent feil',
+        });
+        return;
+      }
+      /* Create QR document */
+      try {
+        res = await createQR(event.eid);
```

Comment:
I was thinking if we want a separate `createQR`🤔 what about the `getQR` just handling creating it if it doesn't exist. This way, the client don't need to call create it first🤷‍♂️

Maybe there is a reason I can't think about that makes that a bad idea though, but that would atleast also simplify this logic

However, I am perfectly ok with that being a fix for later and merging as is to test out for thursday

Reply:
Personally I think get should only concern about fetching the resource and it should be up to the users to create the resource

Reply:
Makes sense to just leave it as is as well, and it's also probably more REST'y